adr/2023_02_27_cache_settings.md

```diff
@@ -0,0 +1,109 @@
+# Cache Settings
+
+> `Status:`
+
+> `Date: 2023-02-27`
+
+> `Participants: Brice Schaffner, Christoph Böcklin`
+
+## Context
+
+`service-stac` uses currently a static cache settings for all requests:
+
+- API requests: 600s / 10min (configured per staging)
+- Assets download: 7200s / 2h (configured per staging)
+
+The assets download cache setting is set when uploading the asset on S3.
+
+For coming new functionality we need to have different cache settings depending at least on the collection,
+even maybe on more fine grained selection (items/assets).
+
+For the diemo service which now uses STAC and updates its assets every minutes, see below, we need to change
+the asset download cache setting ASAP (before March 2023).
+
+### Up coming scenarios
+
+1. Realtime assets, changed every minutes or less (e.g. Diemo)
+2. Very frequent changes, changed every 10 minutes or even less (Meteoschweiz)
+3. Frequent asset updates, (e.g. every hours, twice a day, daily, bi weekly, weekly)
+4. Infrequent updates: here it can be either a define low frequency (e.g. every 2 months, or every year) or sporadic update.
+   1. After update the cache should be invalidated as the updated data must be directly available
+   2. After update we can afford to wait for the cached data expiration.
+5. Never
+
+In theory scenario 5. doesn't exists as potentially every assets might be somewhen updated. Currently we support
+scenario 4. and 5.
+
+### Important facts
+
+- S3 cache settings (`Cache-Control` header) cannot be changed once set, unless the object is uploaded again. Possibility to copy in place.
+- S3 implements pre-conditional requests
+- HTTP `Expires` header is discouraged to be used by AWS: [We recommend that you use the Cache-Control max-age directive instead of the Expires header field to control object caching. If you specify values both for Cache-Control max-age and for Expires, CloudFront uses only the value of Cache-Control max-age](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/Expiration.html#expiration-individual-objects)
+- HTTP `Expires` header is ignored by most if not all implementation if `Cache-Control` with `max-age` or `s-maxage` is present
+- No STAC API specifications for caching management
+- STAC Specification extension: [Timestamps Extension Specification](https://github.com/stac-extensions/timestamps)
+  | Field  | Type   | Description   |
+  |-------------- | -------------- | -------------- |
+  | published    | string     | Date and time the corresponding data (see below) was published the first time, in UTC. |
+  | expires | string | Date and time the corresponding data (see below) expires (is not valid any longer), in UTC. |
+  | unpublished | string | Date and time the corresponding data (see below) was unpublished, in UTC. |
+  
+## Proposal #1
+
+A proposal is to add a new field in the transactional API on the create asset; `update_rate`
```

Comment:
A very picky comment follows, sorry for that ;-)

Calling things `rate` would imply something in the form of  `something per seconds`, so e.g. one update every 60 seconds as rate would be `1/60s` so the unit would be `1/s`, not `s`.

If we provide a value in the form of "number of seconds", why don't we call it `livetime` instead of `rate`

[picky mode off ;-)]

Reply:
@hansmannj what aboute `update_frequence` ?